rosetta/services/transaction_parser.go

```diff
@@ -0,0 +1,244 @@
+package services
+
+import (
+	"math/big"
+
+	"github.com/ElrondNetwork/elrond-go/data/transaction"
+	"github.com/ElrondNetwork/elrond-proxy-go/data"
+	"github.com/ElrondNetwork/elrond-proxy-go/rosetta/client"
+	"github.com/ElrondNetwork/elrond-proxy-go/rosetta/configuration"
+	"github.com/coinbase/rosetta-sdk-go/types"
+)
+
+type transactionsParser struct {
+	config        *configuration.Configuration
+	networkConfig *client.NetworkConfig
+}
+
+func newTransactionParser(cfg *configuration.Configuration, networkConfig *client.NetworkConfig) *transactionsParser {
+	return &transactionsParser{
+		config:        cfg,
+		networkConfig: networkConfig,
+	}
+}
+
+func (tp *transactionsParser) parseTxsFromHyperBlock(hyperBlock *data.Hyperblock) []*types.Transaction {
+	txs := make([]*types.Transaction, 0)
+	for _, eTx := range hyperBlock.Transactions {
+		tx, ok := tp.parseTx(eTx, false)
+		if !ok {
+			continue
+		}
+		txs = append(txs, tx)
+	}
+
+	return txs
+}
+
+func (tp *transactionsParser) parseTx(eTx *data.FullTransaction, isInPool bool) (*types.Transaction, bool) {
+	switch eTx.Type {
+	case string(transaction.TxTypeNormal):
+		return tp.createRosettaTxFromMoveBalance(eTx, isInPool), true
+	case string(transaction.TxTypeReward):
+		return tp.createRosettaTxFromReward(eTx), true
+	case string(transaction.TxTypeUnsigned):
+		return tp.createRosettaTxFromUnsignedTx(eTx)
+	case string(transaction.TxTypeInvalid):
+		return tp.createRosettaTxFromInvalidTx(eTx), true
+	default:
+		return nil, false
+	}
+}
+
+func (tp *transactionsParser) createRosettaTxFromUnsignedTx(eTx *data.FullTransaction) (*types.Transaction, bool) {
+	if eTx.Value == "0" {
+		return nil, false
+	}
+
+	return &types.Transaction{
+		TransactionIdentifier: &types.TransactionIdentifier{
+			Hash: eTx.Hash,
+		},
+		Operations: []*types.Operation{
+			{
+				OperationIdentifier: &types.OperationIdentifier{
+					Index: 0,
+				},
+				Type:   opScResult,
+				Status: OpStatusSuccess,
+				Account: &types.AccountIdentifier{
+					Address: eTx.Receiver,
```

Comment:
todo: check this if we need also a sender with value subtraction

Reply:
An unsigned transaction is a smart contract result that return money to receiver address. So we have only one operation.